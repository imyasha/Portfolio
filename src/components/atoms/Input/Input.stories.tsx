import { Input, InputProps } from './Input';

export default {
	title: 'atoms/Input',
	component: Input,
};

export const InputStory = (args: InputProps) => <Input {...args} />;

InputStory.storyName = 'Input';
InputStory.args = {
	label: 'Input field',
	placeholder: 'Input field placeholder',
	id: 'some-id',
	type: 'text',
};
