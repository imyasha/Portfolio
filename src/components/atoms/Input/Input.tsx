import classNames from 'classnames';
import { Field, useField } from 'formik';

export interface InputProps {
	label: string;
	placeholder?: string;
	id: string;
	type?: 'text' | 'number' | 'password' | 'email' | 'tel';
}

export const Input = ({
	label,
	placeholder,
	id,
	type = 'text',
}: InputProps) => {
	const fieldProps = { label, id, name: id, placeholder, type };
	// eslint-disable-next-line @typescript-eslint/no-unused-vars
	const [_, meta] = useField(fieldProps);

	return (
		<label htmlFor="firstName" className="flex flex-col font-bold mb-2 mt-6">
			{label}
			<Field
				placeholder={placeholder}
				name={id}
				id={id}
				type={type}
				className={classNames(
					'px-4 py-2 rounded-md my-2 ring-1 ring-grey-400 dark:ring-grey-700',
					{
						'ring-red-600': meta.touched && meta.error,
					}
				)}
			/>
			<div className="font-sm font-normal text-red-600">
				{meta.touched && meta.error}
			</div>
		</label>
	);
};
