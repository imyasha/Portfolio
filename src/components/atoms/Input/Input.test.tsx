import { Input } from '../Input';
import { render } from '@testing-library/react';

import { Formik } from 'formik';

const mockField = {
	value: '',
	checked: false,
	onChange: jest.fn(),
	onBlur: jest.fn(),
	multiple: undefined,
	name: 'password',
};
const mockMeta = {
	touched: false,
	error: '',
	initialError: '',
	initialTouched: false,
	initialValue: '',
	value: '',
};

jest.mock('formik', () => ({
	...jest.requireActual('formik'),
	useField: jest.fn(() => {
		return [mockField, mockMeta];
	}),
}));

describe('Input', () => {
	it('renders correctly', () => {
		const { container } = render(
			<Formik
				initialValues={{
					firstName: '',
				}}
				onSubmit={jest.fn()}
				validate={(values) => {
					const errors = {
						firstName: '',
					};

					if (!values?.firstName) {
						errors.firstName = 'Required.';
					}

					return errors;
				}}
			>
				<Input
					label="Some input"
					id="some-input"
					placeholder="Some placeholder"
				/>
			</Formik>
		);

		expect(container).toMatchSnapshot();
	});
});
