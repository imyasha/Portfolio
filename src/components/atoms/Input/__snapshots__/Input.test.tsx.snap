// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders correctly 1`] = `
<div>
  <label
    class="flex flex-col font-bold mb-2 mt-6"
    for="firstName"
  >
    Some input
    <input
      class="px-4 py-2 rounded-md my-2 ring-1 ring-grey-400 dark:ring-grey-700"
      id="some-input"
      name="some-input"
      placeholder="Some placeholder"
      type="text"
      value=""
    />
    <div
      class="font-sm font-normal text-red-600"
    />
  </label>
</div>
`;